// @tanstack/start-plugin-core@1.169.6/dist/esm/rsbuild/start-router-plugin.js
import { routesManifestPlugin } from "../start-router-plugin/generator-plugins/routes-manifest-plugin.js";
import { prerenderRoutesPlugin } from "../start-router-plugin/generator-plugins/prerender-routes-plugin.js";
import { buildRouteTreeFileFooterFromConfig } from "../start-router-plugin/route-tree-footer.js";
import { RSBUILD_ENVIRONMENT_NAMES } from "./planning.js";
import path from "pathe";
//#region src/rsbuild/start-router-plugin.ts
/**
* Registers the TanStack Router generator and code-splitter plugins
* as rspack plugins via `modifyRspackConfig`.
*
* The router-plugin package exports rspack-compatible unplugin wrappers:
* - TanStackRouterGeneratorRspack: file-based route generation
* - TanStackRouterCodeSplitterRspack: route code splitting
*
* These are dynamically imported to avoid hard dependency on router-plugin/rspack.
*/
function registerRouterPlugins(api, opts) {
	api.modifyRspackConfig(async (config, utils) => {
		const envName = utils.environment.name;
		const { startConfig } = opts.getConfig();
		const routerConfig = startConfig.router;
		if (envName === RSBUILD_ENVIRONMENT_NAMES.client) try {
			const { TanStackRouterGeneratorRspack } = await import("@tanstack/router-plugin/rspack");
			const generatorPlugin = TanStackRouterGeneratorRspack({
				...routerConfig,
				target: opts.corePluginOpts.framework,
				routeTreeFileFooter: () => {
					return buildRouteTreeFileFooterFromConfig({
						generatedRouteTreePath: path.resolve(routerConfig.generatedRouteTree),
						getConfig: opts.getConfig,
						corePluginOpts: opts.corePluginOpts
					});
				},
				plugins: [routesManifestPlugin(), ...opts.startPluginOpts?.prerender?.enabled === true ? [prerenderRoutesPlugin()] : []]
			});
			utils.appendPlugins(generatorPlugin);
		} catch {}
		if (envName === RSBUILD_ENVIRONMENT_NAMES.client || envName === RSBUILD_ENVIRONMENT_NAMES.server) {
			const isClient = envName === RSBUILD_ENVIRONMENT_NAMES.client;
			try {
				const { TanStackRouterCodeSplitterRspack } = await import("@tanstack/router-plugin/rspack");
				const splitterPlugin = TanStackRouterCodeSplitterRspack({
					...routerConfig,
					target: opts.corePluginOpts.framework,
					codeSplittingOptions: {
						...routerConfig.codeSplittingOptions,
						deleteNodes: isClient ? [
							"ssr",
							"server",
							"headers"
						] : void 0,
						addHmr: isClient
					}
				});
				utils.appendPlugins(splitterPlugin);
			} catch {}
		}
	});
}
//#endregion
export { registerRouterPlugins };
